src/irmin/node.ml

```diff
@@ -94,13 +98,12 @@ struct
 
   let list ?(offset = 0) ?length (t : t) =
     let take_length seq =
-      match length with None -> seq | Some n -> take n seq
+      match length with None -> List.of_seq seq | Some n -> take n seq
     in
     StepMap.to_seq t
     |> drop offset
     |> take_length
-    |> Seq.map (fun (_, e) -> of_entry e)
-    |> List.of_seq
+    |> List.map (fun (_, e) -> of_entry e)
```

Comment:
Can you keep this as a `Seq.map` to avoid stack overflows

Reply:
Good point, thanks. We should test the case of nodes wider than the stack limit, too.